tests/e2e-targetallocator/targetallocator-kubernetessd/00-install.yaml

```diff
@@ -0,0 +1,133 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ta
+automountServiceAccountToken: true
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: collector
+automountServiceAccountToken: true
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: ta
+rules:
+- apiGroups: [""]
+  resources:
+    - pods
+    - nodes
+    - services
+    - endpoints
+    - configmaps
+    - secrets
+    - namespaces
+  verbs:
+    - get
+    - watch
+    - list
+- apiGroups: ["apps"]
+  resources:
+    - statefulsets
+    - daemonsets
+    - services
+    - endpoints
+  verbs:
+    - get
+    - watch
+    - list
+- apiGroups: ["discovery.k8s.io"]
+  resources:
+    - endpointslices
+  verbs:
+    - get
+    - watch
+    - list
+- apiGroups: ["networking.k8s.io"]
+  resources:
+    - ingresses
+  verbs:
+    - get
+    - watch
+    - list
+- nonResourceURLs: ["/metrics"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: collector
+rules:
+- apiGroups: [""]
+  resources:
+    - pods
+    - nodes
+    - nodes/metrics
+    - services
+    - endpoints
+  verbs:
+    - get
+    - watch
+    - list
+- apiGroups: ["networking.k8s.io"]
+  resources:
+    - ingresses
+  verbs:
+    - get
+    - watch
+    - list
+- nonResourceURLs: ["/metrics", "/metrics/cadvisor"]
+  verbs: ["get"]
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+  - command: kubectl create clusterrolebinding ta-$NAMESPACE --clusterrole=ta --serviceaccount=$NAMESPACE:ta
+  - command: kubectl create clusterrolebinding collector-$NAMESPACE --clusterrole=collector --serviceaccount=$NAMESPACE:collector
+---
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: prometheus-kubernetessd
+spec:
+  mode: daemonset
+  serviceAccount: collector
+  targetAllocator:
+    enabled: true
+    serviceAccount: ta
+    image: "local/opentelemetry-operator-targetallocator:e2e"
+    prometheusCR:
+      enabled: false
```

Comment:
I may be missing something but don't we want to test the new strategy here? Wouldn't our final teststep in 01-install.yaml fail without installing the TA?

Reply:
wouldn't we expect [this](https://github.com/open-telemetry/opentelemetry-operator/pull/2430/files#diff-969909de4d71fcaae234fd2a1ce3bd7c03b726126e2443bb7dc04c8a5145b97eR49) to fail then? And also for [this target allocator](https://github.com/open-telemetry/opentelemetry-operator/pull/2430/files#diff-fbf022ca14c8b039493f038feac0716db643ac051a41ccb76bac47cb3bcefeb8R35-R38) section to be empty?

Reply:
should we require that daemonset mode has to run with the new strategy? right now we require the new strategy has daemonset but not the reverse. I don't think anything could go horribly wrong if you run with least-weighted and daemonset, but i would worry that too much node cycling would cause poorly balanced targets.

Reply:
I added a validation for this as well